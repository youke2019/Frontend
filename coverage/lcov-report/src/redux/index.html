<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/redux</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> src/redux
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">44.05% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>37/84</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">36% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>9/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>36/72</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="actions.js"><a href="actions.js.html">actions.js</a></td>
	<td data-value="69.7" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 69%;"></div><div class="cover-empty" style="width:31%;"></div></div></td>
	<td data-value="69.7" class="pct medium">69.7%</td>
	<td data-value="33" class="abs medium">23/33</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="9.09" class="pct low">9.09%</td>
	<td data-value="11" class="abs low">1/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file low" data-value="reducers.js"><a href="reducers.js.html">reducers.js</a></td>
	<td data-value="24.49" class="pic low"><div class="chart"><div class="cover-fill" style="width: 24%;"></div><div class="cover-empty" style="width:76%;"></div></div></td>
	<td data-value="24.49" class="pct low">24.49%</td>
	<td data-value="49" class="abs low">12/49</td>
	<td data-value="36" class="pct low">36%</td>
	<td data-value="25" class="abs low">9/25</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="48" class="abs low">12/48</td>
	</tr>

<tr>
	<td class="file high" data-value="state.js"><a href="state.js.html">state.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Aug 01 2019 01:16:28 GMT+0800 (GMT+08:00)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
<script src="..\..\block-navigation.js"></script>
</body>
</html>
