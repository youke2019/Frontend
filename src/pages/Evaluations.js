import React from "react";
import {
    Text,
    View,
    ImageBackground,
    StyleSheet,
    TouchableOpacity,
    ScrollView,
} from "react-native"
import EvaluationCard from '../components/EvaluationCard'
import Carousel from 'react-native-snap-carousel'
import axios from "axios"
import {connect} from "react-redux"
import StackNavBar from "../components/StackNavBar"

const mapStateToProps = state => {
    return {
        user: state.user_info
    }
}

class Evaluations extends React.Component {
    state={
        evaluations: [],
        cardScrollEnable : true,
    }

    componentWillMount() {
        this.props.navigation.addListener(
            'willFocus',
            () => {this.getEvaluations()}
        )

        this.getEvaluations()
    }

    getEvaluations = () => {
        axios.get(baseUrl+'/courses/evaluates/find',{
            params:{
                course_id: this.props.navigation.state.params.course_info.course_id,
            }
        }).then(res=>{
            this.setState({
                evaluations: res.data
            })
        }).catch(err => {
            console.log(err)
        })
    }

    changeCardScroll = () => {
        this.setState({
            cardScrollEnable: !this.state.cardScrollEnable,
        })
    }

    postEvaluation = () => {
        this.props.navigation.navigate("NewEvaluation", {
            course_info: this.props.navigation.state.params.course_info,
            user_id: this.props.user.id
        })
    }


    _renderItem = ({item,index}) => {
        return (
            <EvaluationCard
                evaluation={item}
                onDetail={this.changeCardScroll}
            />
        )
    }

    render () {
        const {
            evaluations,
            cardScrollEnable,
        } = this.state

        return (
            <ScrollView
                style={styles.container}
            >
                <StackNavBar
                    navigation={this.props.navigation}
                />
                <View style={styles.header}>
                    <Text style={styles.title}>课程评测</Text>
                </View>
                <TouchableOpacity
                    style={styles.plus_container}
                    onPress={this.postEvaluation}
                >
                    <ImageBackground
                        style={styles.plus_button}
                        imageStyle={{resizeMode: 'stretch'}}
                        source={{uri:'button_orange'}}
                    >
                        <Text style={styles.plus_text}>发布我的评测</Text>
                    </ImageBackground>
                </TouchableOpacity>
                <View
                    style={styles.slider_container}
                >
                    {
                        evaluations.length?
                            <Carousel
                                ref={(c) => { this._carousel = c; }}
                                data={evaluations}
                                renderItem={this._renderItem}
                                sliderWidth={350}
                                itemWidth={300}
                                layout={'stack'}
                                firstItem={evaluations.length-1}
                                layoutCardOffset={10}
                                scrollEnabled={cardScrollEnable}
                                contentContainerCustomStyle={styles.slider}
                            />
                            :
                            null
                    }
                </View>
            </ScrollView>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor:'#FDD32A',
    },
    header:{
        height: 90,
        justifyContent: 'center',
        alignItems: 'center',
    },
    title:{
        color: '#000000',
        fontSize: 40,
        fontFamily: '字魂95号-手刻宋'
    },
    slider_container:{
        justifyContent: 'center',
        alignItems: 'center',
    },
    slider:{
        padding: 30,
    },
    plus_container:{
        justifyContent: 'center',
        alignItems: 'center',
    },
    plus_button:{
        width: 180,
        height: 40,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
    },
    plus_text:{
        paddingHorizontal: 5,
        fontSize: 16,
        letterSpacing: 1,
        fontFamily: '字魂95号-手刻宋',
    }
})

export default connect(
    mapStateToProps,
)(Evaluations)