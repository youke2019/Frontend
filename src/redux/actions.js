export const UPDATE_USER_INFO = 'UPDATEUSERINFO'
export const LOAD_USER_INFO = 'LOADUSERINFO'

export const updateUserInfo = (data) => ({
    type:UPDATE_USER_INFO,
    data
})

export const loadUserInfo = (data) => ({
    type:LOAD_USER_INFO,
    data
})

