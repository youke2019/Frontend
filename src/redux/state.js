const initialState = {
    user_info: null,
    course_list: null,
    login_ready:false,
}

export default initialState