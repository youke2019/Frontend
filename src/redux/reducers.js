import { combineReducers } from 'redux'
import {
    UPDATE_USER_INFO,
    LOAD_USER_INFO,
    CLEAR_USER_INFO,
    SEARCH_COURSES,
} from "./actions.js"
import initialState from "./state";
import axios from 'axios'


function userInfoReducer(state=initialState.user_info, action) {
    switch (action.type) {
        case UPDATE_USER_INFO:
            storage.save({
                key: 'user',
                data: action.data
            })
            state = action.data
            return state
        case LOAD_USER_INFO:
            state = action.data
            return state
        case CLEAR_USER_INFO:
            state = null
            return state
        default:
            return state
    }
}

function courseReducer(state=initialState.course_list, action) {
    switch (action.type) {
        case SEARCH_COURSES:
            state = action.data
            return state
        default:
            return state
    }
}

export const combinedReducer = combineReducers({
    user_info : userInfoReducer,
    course_list: courseReducer,
})