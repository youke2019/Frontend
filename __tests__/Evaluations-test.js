import React from 'react'
import renderer from 'react-test-renderer'
import {shallow, mount, render, configure} from 'enzyme'
import Evaluations from '../src/pages/Evaluations'
import Adapter from "enzyme-adapter-react-16/build"
import {createStore} from "redux";
import {combinedReducer} from "../src/redux/reducers";

configure({adapter: new Adapter()})

jest.mock('axios')
global.baseUrl = 'baseUrl'

let callback=jest.fn()

const navigation = {
    state:{
        params:{
            course_info:{
                course_id: "test",
            }
        }
    },
    navigate: jest.fn(()=>{}),
    addListener: jest.fn((type,func)=>{
        callback = func
    }),
}

const testInitialState = {
    user_info: {
        id: 'testId'
    },
    course_list: null
}

const errorInitialState = {
    user_info: {
        id: 'errorId'
    },
    course_list: null
}

const store = createStore(combinedReducer,testInitialState)
const errorStore = createStore(combinedReducer,errorInitialState)

describe('functions', () => {
    let wrapper = shallow(
        <Evaluations
            store={store}
            navigation={navigation}
        />
    )

    wrapper = wrapper.dive().dive()
    expect(navigation.addListener).toHaveBeenCalled()

    it('getEvaluations() normal', (done) => {
        const promise = new Promise(
            function (resolve, reject)
            {resolve('success')})
        promise
            .then(data=>{
                expect(wrapper.instance().state.evaluations).toHaveLength(1)
                done()
            })
            .catch(err=>{console.log(err)})
    })

    it('changeCardScroll() normal', (done) => {
        wrapper.setState({
            cardScrollEnable: true
        })
        wrapper.instance().changeCardScroll()
        const promise = new Promise(
            function (resolve, reject)
            {resolve('success')})
        promise
            .then(data=>{
                expect(wrapper.instance().state.cardScrollEnable).toBe(false)
                done()
            })
            .catch(err=>{console.log(err)})
    })

    it('postEvaluation()', () => {
        wrapper.instance().postEvaluation()
        expect(navigation.navigate).toHaveBeenCalled()
    })

    it('callback()', () => {
        callback()
    })
})

test('renders Evaluations page correctly', (done) => {
    let wrapper = shallow(
        <Evaluations
            store={errorStore}
            navigation={navigation}
        />
    )
    const promise = new Promise(
        function (resolve, reject)
        {resolve('success')})
    promise
        .then(data=>{
            done()
        })
        .catch(err=>{console.log(err)})
})



test('renders Evaluations page correctly', (done) => {
    const tree = renderer.create(
        <Evaluations
            store={store}
            navigation={navigation}
        />).toJSON()
    const promise = new Promise(
        function (resolve, reject)
        {resolve('success')})
    promise
        .then(data=>{
            expect(tree).toMatchSnapshot()
            done()
        })
        .catch(err=>{console.log(err)})
})