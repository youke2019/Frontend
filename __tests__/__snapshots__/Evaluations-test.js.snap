// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Evaluations page correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#FDD32A",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 10,
          "paddingTop": 30,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 0.7,
            "height": 20,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "navigate-return",
            }
          }
          style={
            Object {
              "height": 25,
              "width": 25,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 5,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "flex": 1,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
           课程
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 90,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "字魂95号-手刻宋",
            "fontSize": 40,
          }
        }
      >
        课程评测
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "width": 180,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "button_orange",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 40,
                "width": 180,
              },
              Object {
                "resizeMode": "stretch",
              },
            ]
          }
        />
        <Text
          style={
            Object {
              "fontFamily": "字魂95号-手刻宋",
              "fontSize": 16,
              "letterSpacing": 1,
              "paddingHorizontal": 5,
            }
          }
        >
          发布我的评测
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    />
  </View>
</RCTScrollView>
`;
